internal/lsp/server.go

```diff
@@ -813,9 +851,8 @@ func (l *LanguageServer) StartTemplateWorker(ctx context.Context) {
 
 			// finally, trigger a diagnostics run for the new file
 			updateEvent := fileUpdateEvent{
-				Reason:  "internal/templateNewFile",
-				URI:     fileURI,
-				Content: newContents,
+				Reason: "internal/templateNewFile",
```

Comment:
The actual issue this fixes was triggered by 

https://github.com/StyraInc/regal/pull/1100/files#diff-0077cc8440363b2137f8e3e405549e59083242b4281d50b08702a6b7153492a8R751-R756

Where an empty update was sent causing the cache's state to be set to blank for files when they were detected as new, despite them first having their contents loaded from disk.